typespec-extension/src/external-schemas.ts

```diff
@@ -108,30 +109,73 @@ export function createPollOperationDetailsSchema(schemas: Schemas, stringSchema:
 
 const fileDetailsMap: Map<string, ObjectSchema> = new Map();
 
+function getFileSchemaName(baseName: string) {
+  return pascalCase(baseName) + (baseName.toLocaleLowerCase().endsWith("file") ? "Details" : "FileDetails");
+}
+
+function createFileDetailsSchema(schemaName: string, propertyName: string, namespace: string, schemas: Schemas) {
+  const fileDetailsSchema = new ObjectSchema(schemaName, 'The file details for the "' + propertyName + '" field.', {
+    language: {
+      default: {
+        namespace: namespace,
+      },
+      java: {
+        namespace: getJavaNamespace(namespace),
+      },
+    },
+    serializationFormats: [KnownMediaType.Multipart],
+  });
+  fileDetailsSchema.serializationFormats;
+  schemas.add(fileDetailsSchema);
+  fileDetailsMap.set(schemaName, fileDetailsSchema);
+  return fileDetailsSchema;
+}
+
 export function getFileDetailsSchema(
-  filePropertyName: string,
+  property: SdkBodyModelPropertyType,
   namespace: string,
   schemas: Schemas,
   binarySchema: BinarySchema,
   stringSchema: StringSchema,
+  processSchemaFunc: (type: SdkType) => Schema,
 ): ObjectSchema {
-  const schemaName =
-    pascalCase(filePropertyName) + (filePropertyName.toLocaleLowerCase().endsWith("file") ? "Details" : "FileDetails");
-  let fileDetailsSchema = fileDetailsMap.get(schemaName);
-  if (!fileDetailsSchema) {
-    fileDetailsSchema = new ObjectSchema(schemaName, 'The file details for the "' + filePropertyName + '" field.', {
-      language: {
-        default: {
-          namespace: namespace,
-        },
-        java: {
-          namespace: getJavaNamespace(namespace),
-        },
-      },
-      serializationFormats: [KnownMediaType.Multipart],
-    });
-    fileDetailsSchema.serializationFormats;
-    schemas.add(fileDetailsSchema);
+  if (property.type.kind === "model") {
+    // property.type is File, use name and properties from property.type for the File schema
+    /*
+    Current logic:
+    - Class name suffix "FileDetails"
+    - No class hierachy for File
+    - File has 3 properties: "content", "filename", "contentType" (Note that it is "contents" in TypeSpec)
+    - No adjustment on "content" property, it is always BinaryData and required
+    - Allow constant type for "filename" and "contentType" (to be discussed for other types e.g. enum)
+    - Allow required for "filename" and "contentType"
+     */
```

Comment:
I summarize current logic here. We would have more discussion in Scrum.

Reply:
Though I have certain logic (above) to handle some complex case, but the logic is not finalized, the goal of this PR is to allow service write basic `@multipartBody` and get good code generated, e.g.

```ts
  @multipartBody body: {
    file: HttpPart<File>
  }
```

and still generate reasonable code on complex cases.

Reply:
The rationale for
- Class name suffix "FileDetails"
is allow dev write `HttpPart<File>` and generate `FileDetails`

If we use the model name directly, user would need to do a `@clientName(File, "FileDetails", "java")`